validator/client/validator.go

```diff
@@ -950,150 +952,169 @@ func (v *validator) PushProposerSettings(ctx context.Context, km keymanager.IKey
+func (v *validator) buildSignedRegReqs(ctx context.Context, pubkeys [][fieldparams.BLSPubkeyLength]byte, signer iface.SigningFunc) ([]*ethpb.SignedValidatorRegistrationV1, error) {
+	var signedValRegRegs []*ethpb.SignedValidatorRegistrationV1
+
+	for i, k := range pubkeys {
+		feeRecipient := common.HexToAddress(params.BeaconConfig().EthBurnAddressHex)
+		gasLimit := params.BeaconConfig().DefaultBuilderGasLimit
+		enabled := false
+		if v.ProposerSettings.DefaultConfig != nil {
+			feeRecipient = v.ProposerSettings.DefaultConfig.FeeRecipient // Use cli config for fee recipient.
+			config := v.ProposerSettings.DefaultConfig.BuilderConfig
+			if config != nil && config.Enabled {
+				gasLimit = config.GasLimit // Use cli config for gas limit.
+				enabled = true
 			}
-			request, err := v.SignValidatorRegistrationRequest(ctx, signer, unsignedRequest)
-			if err != nil {
-				//error is logged and skips appending
-				continue
+		}
+		if v.ProposerSettings.ProposeConfig != nil {
+			config, ok := v.ProposerSettings.ProposeConfig[k]
+			if ok && config != nil {
+				feeRecipient = config.FeeRecipient // Use file config for fee recipient.
+				builderConfig := config.BuilderConfig
+				if builderConfig != nil {
+					if builderConfig.Enabled {
+						gasLimit = builderConfig.GasLimit // Use file config for gas limit.
+						enabled = true
+					} else {
+						enabled = false // Custom config can disable validator from register.
+					}
+				}
 			}
-			signedRegisterValidatorRequests = append(signedRegisterValidatorRequests, request)
+		}
+		if !enabled {
+			continue
+		}
+		req := &ethpb.ValidatorRegistrationV1{
+			FeeRecipient: feeRecipient[:],
+			GasLimit:     gasLimit,
+			Timestamp:    uint64(time.Now().UTC().Unix()),
+			Pubkey:       pubkeys[i][:],
+		}
+		signedReq, err := v.SignValidatorRegistrationRequest(ctx, signer, req)
+		if err != nil {
+			log.WithFields(logrus.Fields{
+				"pubkey":       fmt.Sprintf("%#x", req.Pubkey),
+				"feeRecipient": feeRecipient,
+			}).Error(err)
+			continue
+		}
+		signedValRegRegs = append(signedValRegRegs, signedReq)
+		if hexutil.Encode(feeRecipient.Bytes()) == params.BeaconConfig().EthBurnAddressHex {
+			log.WithFields(logrus.Fields{
+				"pubkey":       fmt.Sprintf("%#x", req.Pubkey),
+				"feeRecipient": feeRecipient,
+			}).Warn("Fee recipient is burn address")
 		}
 	}
-
-	if len(proposerFeeRecipientRequests) != len(pubkeys) {
-		log.WithFields(logrus.Fields{
-			"totalNonActivePubkeys": len(pubkeys) - len(proposerFeeRecipientRequests),
-		}).Warnln("will not prepare beacon proposer and update fee recipient until a validator index is assigned")
-	}
-
-	if len(signedRegisterValidatorRequests) != len(pubkeys) && anyValidatorRegistrationEnabled {
-		log.WithFields(logrus.Fields{
-			"totalNonActivePubkeys": len(pubkeys) - len(signedRegisterValidatorRequests),
-		}).Warnln("will not be included in builder validator registration until a validator index is assigned")
-	}
```

Comment:
I completely removed this warning log. It feels it'd be prone to miss information. It's normal for `len(pubkeys) != len(req)` given any validator could be `enabled: false`